src/state/migrations.ts

```diff
@@ -0,0 +1,69 @@
+import { createMigrate, MigrationManifest, PersistedState, PersistMigrate } from 'redux-persist'
+import { MigrationConfig } from 'redux-persist/es/createMigrate'
+
+import { legacyTransactionMigrations, legacyUserMigrations } from './legacyMigrations'
+import { initialState as initialListsState } from './lists/reducer'
+import { initialState as initialTransactionsState } from './transactions/reducer'
+import { initialState as initialUserState } from './user/reducer'
+
+function tryParseOldState<T>(value: string | null, fallback: T): T {
+  try {
+    return value ? JSON.parse(value) : fallback
+  } catch (e) {
+    return fallback
+  }
+}
+
+/**
+ * These run once per state re-hydration when a version mismatch is detected.
+ * Keep them as lightweight as possible.
+ *
+ * Migration functions should not assume that any value exists in localStorage previously,
+ * because a user may be visiting the site for the first time or have cleared their localStorage.
+ */
+
+// The target version number is the key
+export const migrations: MigrationManifest = {
+  0: () => {
```

Comment:
This is a legacy migration, not a v0 migration, because it checks localStorage which need only be done when migrating. Can you not put it in the `MigrationManifest`, and instead leave it outside as `legacy`?